web/server/entrypoint.go

```diff
@@ -312,20 +312,22 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 					continue
 				}
 				var options weboptions.Options
-				if !diff.NetworkEqual {
+
+				if !diff.NetworkEqual || !diff.MediaEqual {
+					if err := myRobot.StopWeb(); err != nil {
```

Comment:
I will let someone else comment about changing the order in which StopWeb() is called